packages/app/src/cli/services/app-logs/logs-command/ui/components/hooks/useSelfAdjustingInterval.ts

```diff
@@ -0,0 +1,26 @@
+import {useRef, useEffect} from 'react'
+
+export function useSelfAdjustingInterval<T extends {retryIntervalMs: number | null}>(callback: () => Promise<T>) {
```

Comment:
Nice, I like this abstraction, I like that the whole managing interval is in its own hook now. And maybe the test were flakey due to not having the a `cleanTimeout` 🤔.